alpha/apps/kaltura/lib/kContextDataHelper.php

```diff
@@ -210,23 +205,22 @@ private function setContextDataFlavorAssets($flavorTags)
 
 		$flavorParamsIds = null;
 		$flavorParamsNotIn = false;
-		if(!$this->isAdmin)
-		{
-			foreach ($this->contextDataResult->getActions() as $action) 
-			{	
-				if($action->getType() == RuleActionType::BLOCK)
-				{
-					//in case of block action do not set the list of flavors
-					return;
-				}
-				if($action->getType() == RuleActionType::LIMIT_FLAVORS)
-				{
-					/* @var $action kAccessControlLimitFlavorsAction */
-					$flavorParamsIds = explode(',', $action->getFlavorParamsIds());
-					$flavorParamsNotIn = $action->getIsBlockedList();
-				}
-			}	
-		}
+		
+		foreach ($this->contextDataResult->getActions() as $action) 
```

Comment:
Must validate that contextDataResult is not null, in case of optimized_playback applyContext will not be called.